client/client.go

```diff
@@ -68,56 +73,62 @@ func New(messageService messageservice.MessageService, chainservice chainservice
 	c.receivedVouchers = make(chan payments.Voucher, 1000)
 
 	c.channelNotifier = notifier.NewChannelNotifier(store, c.vm)
+
 	// Start the engine in a go routine
+
 	go c.engine.Run()
 
+	c.wg = &sync.WaitGroup{}
+	c.wg.Add(1)
+	ctx, cancel := context.WithCancel(context.Background())
+	c.cancelEventHandler = cancel
```

Comment:
When we resolve #1341, let's return to this section and add some comments to help the reader understand what is going on.